src/gui/DatabaseWidget.cpp

```diff
@@ -939,23 +927,17 @@ void DatabaseWidget::unlockDatabase(bool accepted)
     }
 }
 
-/**
- * @author Fonic <https://github.com/fonic>
- * Add 'copy-on-doubleclick' functionality for certain columns
- *
- * TODO:
- * If pull request #1298 gets merged, double-clicking column 'Attachments'
- * could open the new details view (see second screenshot)
- */
 void DatabaseWidget::entryActivationSignalReceived(Entry* entry, EntryModel::ModelColumn column)
 {
-    /* Should never happen */
-    if (!entry) {
-        Q_ASSERT(false);
-        return;
-    }
+    Q_ASSERT(entry);
```

Comment:
Don't delete the if block. The assertion will only trigger in debug mode. A following if statement will ensure safe return (punny, isn't it?) in production.

Reply:
That's why I was reluctant to change this. That would mean doubling the expression, which is trivial here, but not so in EntryView.cpp. Doubling expressions is considered bad practice due to increased likeliness of errors, that's why I used Q_ASSERT(false) and most likely why that can be found at numerous places throughout the code.

Not producing a useful error message can't be an issue - you get the file name and line number, any good IDE even takes you right to the location.

Reply:
Q_ASSERT(false) is pretty unhelpful, especially when using Qt which makes usage of debuggers without pretty printers a mess anyway. If the condition is simple and both lines are directly next to each other, I don't see a problem with duplicating it. It also makes sure that CI prints a useful error message, because depending on your system configuration and Qt version, you may not be able to reproduce it locally.

Reply:
Done